cincinnati/src/lib.rs

```diff
@@ -145,6 +146,17 @@ impl<'a> Iterator for PreviousReleases<'a> {
 #[derive(Debug, Clone)]
 pub struct Empty;
 
+/// Errors that can be returned by the methods in this library
+pub mod errors {
+    /// Edge already exists
+    #[derive(Debug, Fail, Eq, PartialEq)]
+    #[fail(display = "edge from {:?} to {:?} already exists", from, to)]
```

Comment:
I'm not sure this is very helpful. Node indexes depend on the specific graph instance/ordering. Should we maybe track source and destination versions instead?

Reply:
I changed it to be a String here so it can be fed a version string at instantiation.